test/test_linalg.py

```diff
@@ -7376,6 +7376,66 @@ def test_permute_matmul(self):
         c = a.permute(0, 1, 3, 2).matmul(b)
         self.assertEqual([c.min(), c.max(), c.sum()], [24, 24, 414720])
 
+
+    @onlyCUDA
+    @unittest.skipIf(not CUDA11OrLater, "Only CUDA 11+ is supported")
+    @toleranceOverride({torch.float32: xtol(atol=1e-5, rtol=1e-5)})
+    @dtypes(torch.float16, torch.bfloat16, torch.float32)
+    @parametrize(
+        "batch_size, N, M, P",
+        [(2, 100, 100, 100),
+         (2, 1000, 1000, 1000),
+         (1, 10000, 1000, 10000),
+         (1, 10000, 10000, 10000)],
+        name_fn=lambda batch_size, N, M, P: "{}_{}_{}_{}".format(batch_size, N, M, P),
+    )
+    def test_cublas_baddbmm_large_input(self, device, batch_size, N, M, P, dtype):
+        torch.manual_seed(23)
+
+        cpu_dtype = dtype
+        if dtype == torch.float16 or dtype == torch.bfloat16:
+            cpu_dtype = torch.float32
+
+        M1 = torch.rand((N, M), device=device, dtype=dtype)
+        M2 = torch.rand((M, P), device=device, dtype=dtype)
+        A = torch.rand((N, P), device=device, dtype=dtype)
+
+        def _convert_to_cpu(t):
+            return t.to(device='cpu', dtype=cpu_dtype)
+        M1_cpu, M2_cpu, A_cpu = map(_convert_to_cpu, [M1, M2, A])
+
+        # linear
+        out1_cpu = torch.nn.functional.linear(M1_cpu, M2_cpu.t(), A_cpu).to(dtype=dtype)
+        out1_gpu = torch.nn.functional.linear(M1, M2.t(), A)
+        self.assertEqual(out1_cpu, out1_gpu.cpu())
+        # test multiply the identity matrix
+        if N == M and M == P:
+            M2_eye = torch.eye(N, device=device, dtype=dtype)
+            out1_eye_gpu = torch.nn.functional.linear(M1, M2_eye.t(), torch.zeros_like(A))
+            self.assertEqual(M1.cpu(), out1_eye_gpu.cpu())
+
+        # baddbmm
+        alpha, beta = 1.0, 1.0
+        M1 = M1.expand(batch_size, N, M)
```

Comment:
Do we want to pass something with a `stride` of zero, or do we want to make it contiguous?

Reply:
Hi @lezcano, what do you mean by `stride` of zero? I'm trying to use the same input for the `linear` and `baddbmm`, so that we can do the cross comparison of the results.

Reply:
Then I don't understand why are you parametrising over `batch_size`, rather than simply doing a `unsqueeze(0)`.

Reply:
I think it might be better to cover cases when batch_size have different values. If batch_size is always 1, then it's equivalent to unsqueeze(0)